### PR TITLE
Replace PTB dataset to Sherlock Holmes 

### PR DESCRIPTION
**Pull request by liyujiel**
## Description ##
Replace PTB dataset to self-created dataset Sherlock Holmes 

Replace reason: https://github.com/dmlc/web-data/pull/75

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [x] files related with Penn TreeBank dataset

## Comments ##
- the model parallel LSTM example cannot run with Penn TreeBank dataset originally, created an issue and wait for contributor update

### CODE REVIEW

File: perl-package/AI-MXNet/examples/get_sherlockholmes_data.sh [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+echo
+echo "NOTE: To continue, you need to review the licensing of the data sets used by this script"
+echo "See https://catalog.ldc.upenn.edu/ldc99t42 for the licensing"
```
Comment (marcoabreu) [inline perl-package/AI-MXNet/examples/get_sherlockholmes_data.sh]: Does this still match for the new dataset?

**Comment by marcoabreu** [comment]
Is the underlying dataset equivalent? I don't think we have test coverage for all examples, so we should make sure that the examples are not broken by making this change.

File: perl-package/AI-MXNet/examples/get_sherlockholmes_data.sh (hunk 1 above)
Comment (liyujiel) [reply perl-package/AI-MXNet/examples/get_sherlockholmes_data.sh]: Thanks for point out! I will change the license to Gutenberg one:
https://www.gutenberg.org/wiki/Gutenberg:The_Project_Gutenberg_License

File: example/rnn/old/README.md [hunk 2]
```diff
@@ -3,7 +3,7 @@ RNN Example
 This folder contains RNN examples using low level symbol interface.
 
 ## Data
-Run `get_ptb_data.sh` to download PenTreeBank data.
+Run `get_sherlockholmes_data.sh` to download PenTreeBank data.
```
Comment (szha) [inline example/rnn/old/README.md]: here

File: example/rnn/old/lstm_sherlockholmes.R [hunk 3]
```diff
@@ -17,7 +17,7 @@
 
 # PennTreeBank Language Model using lstm, you can replace mx.lstm by mx.gru/ mx.rnn to use gru/rnn model
```
Comment (szha) [inline example/rnn/old/lstm_sherlockholmes.R]: here

File: example/rnn/word_lm/train.py [hunk 4]
```diff
@@ -25,7 +25,7 @@
 from mxnet.model import BatchEndParam
 
 parser = argparse.ArgumentParser(description='PennTreeBank LSTM Language Model')
```
Comment (szha) [inline example/rnn/word_lm/train.py]: here

File: example/rnn/bucketing/README.md [hunk 5]
```diff
@@ -3,13 +3,13 @@ RNN Example
 This folder contains RNN examples using high level mxnet.rnn interface.
 
 ## Data
-1) Review the license for the PenTreeBank dataset and ensure that you agree to it. Then uncomment the lines in the 'get_ptb_data.sh' script that download the dataset.
+1) Review the license for the PenTreeBank dataset and ensure that you agree to it. Then uncomment the lines in the 'get_sherlockholmes_data.sh' script that download the dataset.
```
Comment (szha) [inline example/rnn/bucketing/README.md]: update reference to penn tree bank

Comment (szha) [reply example/rnn/bucketing/README.md]: you can search for "PTB" or "tree" to find the rest of the references.

**Pull request merged**